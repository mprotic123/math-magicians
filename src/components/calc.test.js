import React from 'react';
import renderer from 'react-test-renderer';
import { cleanup, fireEvent, render } from '@testing-library/react';
import Calc from './calc.js';

it('Renders calculator', () => {
  const calc = renderer.create(
    <Calc />,
  ).toJSON();
  expect(calc).toMatchSnapshot();
});

it('captures clicks', (done) => {
  function handleClick() {
    done();
  }
  const { getByText } = render(
    <button onClick={handleClick}>Click Me</button>,
  );
  const node = getByText('Click Me');
  fireEvent.click(node);
});

afterEach(cleanup);