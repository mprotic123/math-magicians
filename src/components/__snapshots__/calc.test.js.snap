// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders calculator 1`] = `
<div
  className="calculator"
>
  <div
    className="answer"
  >
    0
  </div>
  <div
    className="buttons"
  >
    <button
      className="button"
      id="AC"
      onClick={[Function]}
      type="button"
    >
      AC
    </button>
    <button
      className="button"
      id="+/-"
      onClick={[Function]}
      type="button"
    >
      +/-
    </button>
    <button
      className="button"
      id="%"
      onClick={[Function]}
      type="button"
    >
      %
    </button>
    <button
      className="button orange"
      id="รท"
      onClick={[Function]}
      type="button"
    >
      รท
    </button>
    <button
      className="button"
      id="7"
      onClick={[Function]}
      type="button"
    >
      7
    </button>
    <button
      className="button"
      id="8"
      onClick={[Function]}
      type="button"
    >
      8
    </button>
    <button
      className="button"
      id="9"
      onClick={[Function]}
      type="button"
    >
      9
    </button>
    <button
      className="button orange"
      id="ร"
      onClick={[Function]}
      type="button"
    >
      x
    </button>
    <button
      className="button"
      id="4"
      onClick={[Function]}
      type="button"
    >
      4
    </button>
    <button
      className="button"
      id="5"
      onClick={[Function]}
      type="button"
    >
      5
    </button>
    <button
      className="button"
      id="6"
      onClick={[Function]}
      type="button"
    >
      6
    </button>
    <button
      className="button orange"
      id="-"
      onClick={[Function]}
      type="button"
    >
      -
    </button>
    <button
      className="button"
      id="1"
      onClick={[Function]}
      type="button"
    >
      1
    </button>
    <button
      className="button"
      id="2"
      onClick={[Function]}
      type="button"
    >
      2
    </button>
    <button
      className="button"
      id="3"
      onClick={[Function]}
      type="button"
    >
      3
    </button>
    <button
      className="button orange"
      id="+"
      onClick={[Function]}
      type="button"
    >
      +
    </button>
    <button
      className="button double-button"
      id="0"
      onClick={[Function]}
      type="button"
    >
      0
    </button>
    <button
      className="button"
      id="."
      onClick={[Function]}
      type="button"
    >
      .
    </button>
    <button
      className="button orange"
      id="="
      onClick={[Function]}
      type="button"
    >
      =
    </button>
  </div>
</div>
`;
